### ISSUE TITLE
Curious about rounding

### ISSUE THREAD

**Issue by JPHutchins** [PetteriAimonen/libfixmath#50]
Often, my use case may be to avoid soft floats on hardware that has no FPU.  See this section of code:

https://github.com/PetteriAimonen/libfixmath/blob/d308e466e1a09118d03f677c52e5fbf402f6fdd0/libfixmath/fix16.h#L72-L89

First, I think it's problematic that "from double" is disregarding the rounding compile def while "from float" respects it.

Second, I'm curious about the statement:
```c
temp += (temp >= 0) ? 0.5f : -0.5f;
```
In my use case, won't this bring in the soft fadd that I don't want?  Then again, I suppose that the user is just doing it wrong if they don't want any floats yet the end up using fix16_from_float at runtime!

**Comment by PetteriAimonen** [comment]
I agree that the rounding difference is weird and should be fixed. Not sure which way makes more sense though.

To avoid any float use, I think you'd use `F16()` or `F16C()` macro. Though probably in most cases compiler will optimize away the inline functions.

**Comment by JPHutchins** [comment]
Totally agreed re: `F16` and `F16C`.
